### ISSUE TITLE
[DEV] Build failing? org.openhab.binding.amazondashbutton

### ISSUE THREAD

**Issue by hreichert** [openhab/openhab2-addons#2722]
Hi everyone,
do I have a local probleme here, or is the build failing?

I'm on openhab/openhab2-addons commit id 17aa3a218cfbfe670bd0b30f2b69995d9da0905b, and latest eclipse/smarthome master and openhab/openhab-core (which builds fine).

Maven error (mvn clean install):

```
[INFO] ------------------------------------------------------------------------
[INFO] Building Amazon Dash Button Binding 2.2.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ org.openhab.binding.amazondashbutton ---
[INFO] Deleting /home/holger/DATA/dev/eclipse4.7_openhab/openhab2-master/git/openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/target
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:build-qualifier (default-build-qualifier) @ org.openhab.binding.amazondashbutton ---
[INFO] The project's OSGi version is 2.2.0.201709201846
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:validate-id (default-validate-id) @ org.openhab.binding.amazondashbutton ---
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:validate-version (default-validate-version) @ org.openhab.binding.amazondashbutton ---
[INFO] 
[INFO] --- maven-resources-plugin:2.4.3:resources (default-resources) @ org.openhab.binding.amazondashbutton ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/holger/DATA/dev/eclipse4.7_openhab/openhab2-master/git/openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/src/main/resources
[INFO] 
[INFO] --- tycho-compiler-plugin:1.0.0:compile (default-compile) @ org.openhab.binding.amazondashbutton ---
[INFO] Compiling 12 source files to /home/holger/DATA/dev/eclipse4.7_openhab/openhab2-master/git/openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/target/classes
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] openHAB 2 Add-ons .................................. SUCCESS [  0.913 s]
[INFO] openHAB Add-ons .................................... SUCCESS [  0.037 s]
[INFO] openHAB Bindings ................................... SUCCESS [  0.035 s]
[INFO] AirQuality Binding ................................. SUCCESS [ 32.029 s]
[INFO] AllPlay Binding .................................... SUCCESS [ 15.123 s]
[INFO] Amazon Dash Button Binding ......................... FAILURE [  0.584 s]
[... more skipped ...]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:39 min
[INFO] Finished at: 2017-09-20T20:48:26+02:00
[INFO] Final Memory: 107M/863M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.0.0:compile (default-compile) on project org.openhab.binding.amazondashbutton: Compilation failure: Compilation failure: 
[ERROR] /home/holger/DATA/dev/eclipse4.7_openhab/openhab2-master/git/openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/src/main/java/org/openhab/binding/amazondashbutton/internal/AmazonDashButtonHandlerFactory.java:[46] 
[ERROR] 	return null;
[ERROR] 	       ^^^^
[ERROR] Null type mismatch: required '@NonNull ThingHandler' but the provided value is null
[ERROR] 1 problem (1 error)
[ERROR] -> [Help 1]
```

If this is a local problem, how can I fix it? Am a bit lost right now...

Thanks!

**Comment by martinvw** [comment]
I just started a build which successfully passes the Amazon Dash Button, do you no 'local' changes on your branch / workspace?

**Comment by martinvw** [comment]
Are you sure you are building with the most up-to-date versions:

```
21:41:34.172  |  [INFO] 
21:41:34.172  |  [INFO] --- tycho-packaging-plugin:1.0.0:build-qualifier (default-build-qualifier) @ org.openhab.binding.amazondashbutton ---
21:41:34.173  |  [INFO] The project's OSGi version is 2.2.0.201709201937
```

Maybe you are connected to the master of your fork's which have not been updated?

**Comment by martinvw** [comment]
> If this is a local problem, how can I fix it? Am a bit lost right now...

Please provide more information about your env, what jdk, what maven, maybe name the commits of the other repo's as well.

**Comment by hreichert** [comment]
> I just started a build which successfully passes the Amazon Dash Button, do you no 'local' changes on your branch / workspace?
https://openhab.ci.cloudbees.com/job/openHAB2-Bundles/153

From the log output of this build:
> 21:37:58.355  |  Executing Maven:  -B -f /scratch/jenkins/workspace/openHAB2-Bundles/pom.xml clean install -DskipChecks=true

Maybe the `-DskipChecks=true` leads to the passing build?

**Comment by martinvw** [comment]
Feel free to test that 😉

**Comment by hreichert** [comment]
Well, no, it's failing for me even with -DskipChecks=true...
hm

**Comment by hreichert** [comment]
See my previous comment https://github.com/openhab/openhab2-addons/issues/2722#issuecomment-330970846
Updated everything and tried again, but no luck. Don't know what to examine further.

**Comment by hreichert** [comment]
OK- it builds now.

See file
`openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/.settings/org.eclipse.jdt.core.prefs`

```
eclipse.preferences.version=1
org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.7
org.eclipse.jdt.core.compiler.compliance=1.7
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=1.7
```

I replaced every occurence of `1.7` with `1.8` and now it works.
Then I grepped trough other org.eclipse.jdt.core.prefs files and found the following, that also have `1.7`:
./org.openhab.binding.tankerkoenig/.settings/org.eclipse.jdt.core.prefs
./org.openhab.binding.max.test/.settings/org.eclipse.jdt.core.prefs
./org.openhab.binding.minecraft/.settings/org.eclipse.jdt.core.prefs
./org.openhab.binding.amazondashbutton/.settings/org.eclipse.jdt.core.prefs
./org.openhab.binding.tellstick/.settings/org.eclipse.jdt.core.prefs
./org.openhab.binding.nest/.settings/org.eclipse.jdt.core.prefs
./org.openhab.binding.hdpowerview/.settings/org.eclipse.jdt.core.prefs
./org.openhab.binding.sleepiq/.settings/org.eclipse.jdt.core.prefs
./org.openhab.binding.kostalinverter/.settings/org.eclipse.jdt.core.prefs
./org.openhab.binding.chromecast/.settings/org.eclipse.jdt.core.prefs

These bindings also fail to build.

As 1.8 is the current level, should I change these prefs and make a PR for it?

**Comment by martinvw** [comment]
Yes please! And if you mention `closes #2722` it will close this issue as soon as it is merged, please also make sure that your commit message includes a `Signed-off-by` (to certify that you wrote it or otherwise have the right to pass it on as an open-source), see http://docs.openhab.org/developers/contributing/contributing.html#sign-your-work

Thanks for sharing!

**Comment by hreichert** [comment]
Sure, here you go:
https://github.com/openhab/openhab2-addons/pull/2724
